hack/run-in-node-container.sh

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+# Copyright 2021 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script runs $@ in a node container
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+REPO_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd -P)"
+cd "${REPO_ROOT}"
+
+NODE_IMAGE="$(cat "${REPO_ROOT}"/.node-image | tr -d '\n')"
+
+docker run \
+    --rm -it \
+    --user $(id -u):$(id -g) \
+    -e HOME=/tmp \
```

Comment:
yarn tries to read $HOME/.yarnrc and put a cache under $HOME/.cache, when we change the user in the line above (so output is correctly owned on the host) the user doesn't exist and yarn fails on inability to open `/.yarnrc`. Setting it to a valid writeable location allows yarn to function and is pretty harmless, yarn will be writing a system cache copy of package on install under `/tmp/.cache`.

The rest of this is hopefully more self-explanatory.

Reply:
should this explanation be a comment above `--user` line?

Reply:
I can do it, but we'll have to abuse a bash trick ;-)

technically normal comments don't work between line continuation (`\`)

Reply:
Add it above `docker run`?

Reply:
I wound up doing that :-)

You can do this for future reference:
```bash
foo \
 `#bar comment` \
  baz
```

This works with backticks (which some shells considered deprecated vs `$()` but realistically cannot ever remove) and does not actually create a subshell. But when I discovered it trying to find a good way to comment a more complex docker invocation in POSIX shell I couldn't find many examples of it in the wild.

In bash the decent answer is to place the args in an array and comment the array instead.

```bash
docker_options = (
    # bar comment
    'foo'
    # quux comment
    'baz'
)
docker run "${docker_options[@]}"
```